Source/Core/Core/Debugger/CodePatch.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/Debugger/CodePatch.h"
+
+#include "Core/PowerPC/PowerPC.h"
+
+CodePatchBase::~CodePatchBase()
+{
+}
+
+CodePatch::CodePatch(u32 address, const std::vector<u8>& patch) : m_address(address), m_patch(patch)
+{
+}
+
+bool CodePatch::Apply()
+{
+  if (GetState() != State::SOURCE)
+    return false;
+
+  for (u32 index = 0, size = static_cast<u32>(m_patch.size()); index < size; index += 1)
```

Comment:
Can you move the declaration of `size` out of that loop? It has nothing to do with the iteration variable `index` and shouldn't be in there.
Also, I believe `++index` is the preferred writing for the increment.